Source/WebCore/platform/graphics/TextTransform.h

```diff
@@ -35,4 +35,34 @@ namespace WebCore {
 // https://w3c.github.io/csswg-drafts/css-text/#full-width
 String transformToFullWidth(const String& text);
 
+std::optional<char32_t> convertToSingleCodePoint(StringView);
+
+// These are the mathvariant values from the MathML recommendation.
+// The special value none means that no explicit mathvariant value has been specified.
+// Note that the numeral values are important for the computation performed in the legacyMathVariant function, do not change them!
+// These values are deprecated with MathML Core.
+enum class MathVariant {
+    None = 0,
+    Normal = 1,
+    Bold = 2,
+    Italic = 3,
+    BoldItalic = 4,
+    Script = 5,
+    BoldScript = 6,
+    Fraktur = 7,
+    DoubleStruck = 8,
+    BoldFraktur = 9,
+    SansSerif = 10,
+    BoldSansSerif = 11,
+    SansSerifItalic = 12,
+    SansSerifBoldItalic = 13,
+    Monospace = 14,
+    Initial = 15,
+    Tailed = 16,
+    Looped = 17,
+    Stretched = 18
+};
+
+char32_t legacyMathVariant(char32_t codePoint, MathVariant mathvariant);
```

Comment:
The argument names here aren’t helpful. The type `char32_t` is always a "code point" and the type `MathVariant` is always a "math variant".
```suggestion
char32_t legacyMathVariant(char32_t, MathVariant);
```
But also, I can’t find where this new function is called. If this pull request is adding a function that is not yet called anywhere, then maybe we should wait and add the function when it we are also adding the code that calls it.

Reply:
Agreed on the argument names, I have removed them.

`legacyMathVariant` is called in `RenderMathMLToken::updateMathVariantGlyph()`. It is the function that was previously defined in that file. It was previously called `mathVariant`, but I added "legacy" so that it is clear that it is different from the [new `mathvariant`](https://w3c.github.io/mathml-core/#dfn-mathvariant) attribute from MathML Core, which does pretty much the opposite, and that most of it's behaviour is deprecated.

Reply:
I think we can come up with a better name. The word "legacy" seems so vague here.

Reply:
Maybe `mathVariantMapCodePoint`?